Source/WebCore/page/Navigation.h

```diff
@@ -31,10 +31,36 @@
 #include "NavigationHistoryEntry.h"
 #include "NavigationTransition.h"
 #include <JavaScriptCore/JSCJSValue.h>
+#include <wtf/text/StringHash.h>
 
 namespace WebCore {
 
 class HistoryItem;
+class SerializedScriptValue;
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#navigation-api-method-tracker
+struct NavigationAPIMethodTracker : public RefCounted<NavigationAPIMethodTracker> {
```

Comment:
Why is this a reference-counted object instead of just a `struct`?

Reply:
I felt like this is just the nicer option.

I tried `unique_ptr` but that doesn't work for its usage as by spec it will have multiple places referencing it.

It seems unnecessary to ever copy it as a plain object. So I think either its a raw pointer or a RefPtr.

Reply:
That sounds wrong to me. Seems to me that we should be passing and moving an object around, not allocating it on the heap.

Reply:
Ok done.